cmd/kube-controller-manager/app/core.go

```diff
@@ -71,9 +71,23 @@ func startServiceController(ctx ControllerContext) (bool, error) {
 }
 
 func startNodeController(ctx ControllerContext) (bool, error) {
-	_, clusterCIDR, err := net.ParseCIDR(ctx.Options.ClusterCIDR)
-	if err != nil {
-		glog.Warningf("Unsuccessful parsing of cluster CIDR %v: %v", ctx.Options.ClusterCIDR, err)
+	var clusterCIDRs []*net.IPNet
+	for _, cidr := range ctx.Options.ClusterCIDRs {
+		_, cc, err := net.ParseCIDR(cidr)
+		if err != nil {
+			glog.Warningf("Unsuccessful parsing of cluster CIDR %v: %v", ctx.Options.ClusterCIDR[0], err)
+		}
+		clusterCIDRs = append(clusterCIDRs, cc)
+	}
+	NodeCIDRMaskSizes := ctx.Options.NodeCIDRMaskSizes
```

Comment:
From below, it seems like ctx.Options.NodeCIDRMaskSizes is a slice. Do we need to validate that there are as many masks as CIDRs? Do we need to check that they are valid masks? Or is this already done prior to getting here?